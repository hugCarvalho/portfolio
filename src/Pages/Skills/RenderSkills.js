import React, { useState } from "react";
import { useLocation } from "react-router-dom";
import { IsActiveContext } from "../../components/Header/NavButtons/RenderNavButtons";
import useIsDesktop from "../../components/Hooks";
import { server } from "../../config/server";
import { SKILL_CATEGORIES, SKILLS_DATA } from "../../config/Skills";
import FilterBar from "../Projects/FilterBy/FilterBy";
import "./RenderSkills.scss";
import { RelatedSkills, SingleSkill, SingleSkillSvg } from "./SkillCards";

function Skills() {
  const pageIsActive = React.useContext(IsActiveContext);
  const [filterBy, setFilterBy] = useState("all")
  const {pathname} = useLocation()
  const isDesktop = useIsDesktop()

  React.useEffect(()=> {
    fetch(`${server}${pathname}`, {
      method: "POST",
      headers: {
        'Content-Type': 'application/json'
      },
    })
  }, [])

  React.useEffect(() => {
    document.title = "Skills";
  }, [pageIsActive]);

  const skillsData = filterBy === "all" ? SKILLS_DATA : SKILLS_DATA.filter(skill => {
    return filterBy === skill.category
  })
  const hideFilter = pathname === "/skills" && !isDesktop

  return (
    <>
      {!hideFilter && <FilterBar filterBy={filterBy} setFilterBy={setFilterBy} filters={Object.values(SKILL_CATEGORIES)}/>}
      <div className="Skills">
        <main>
          {skillsData.map((skill, i) => {
            if(skill.name === "React Testing Library"){
              return <SingleSkill key={i} classes={skill.class} name={<small>React Testing Library</small>}/>
            }
            if(skill.name === "Supabase"){
              return <SingleSkillSvg key={i} src={skill.src} style={{height: "40%"}}/>
            }
            if(skill.type === "single"){
              return <SingleSkill key={i} classes={skill.class} name={skill.name}/>
            }
            if (skill.type === "singleSvg"){
              return <SingleSkillSvg key={i} src={skill.src} name={skill.name}/>
            }
            if(skill.type === "dual"){
              return <RelatedSkills
                key={i}
                skillOne={skill.name.one}
                classesOne={skill.class.one}
                skillTwo={skill.name.two}
                classesTwo={skill.class.two}
            />
            }
            throw new Error(`skill is not mapped!${skill}`)
          })}
        </main>
      </div>
    </>
  );
}

export default Skills;
