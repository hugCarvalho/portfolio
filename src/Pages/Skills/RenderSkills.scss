.Skills {
  // background-color: #3c6274;
  //background-color: #b3b6b8;
  background-color: #859196;
  background-image: url("./assets/19.png");
  min-height: calc(100vh - 235px); //footer + padding
  display: flex;
  padding: 70px 12% 67px;
  //padding: 87px 12% 60px;

  main {
    width: 100%;
    display: flex;
    flex-wrap: wrap;
    justify-content: center;
  }

  .container__skill-card {
    padding-bottom: 70px;
    margin: auto;
  }

  .wire {
    position: relative;
    width: 100px;
    height: 180px;
    min-height: 155px;
    border-top: 1px solid black;

    //AFFECTS NAIL
    margin: auto;
    margin-bottom: -140px;
    border-radius: 100%;

    display: flex;
    justify-content: center;

    .nail {
      background: gray;
      width: 10px;
      height: 10px;
      margin-top: -5px;
      border-radius: 100%;
      box-shadow: 3px 3px 5px 1px rgb(14, 13, 13);
    }
  }
}

@media (max-width: 920px) {
  .Skills {
    padding-left: 1%;
    padding-right: 1%;
  }
}

// @media (max-width: 622px) {
//   .Skills {
//     min-height: calc(100vh - 245px);
//   }
// }
