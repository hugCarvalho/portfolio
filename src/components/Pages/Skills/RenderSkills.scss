.container__skills {
  background-color: #3c6274;
  min-height: calc(100vh - 270px); //footer + padding
  display: flex;
  padding: 70px 12% 40px;
  align-items: center;

  .wrapper__all {
    width: 100%;
    display: flex;
    flex-wrap: wrap;
  }

  .container__skill-card {
    padding-bottom: 70px;
    margin: auto;
  }

  .wire {
    background-color: lime;
    position: relative;
    width: 100px;
    height: 180px;
    min-height: 155px;
    margin-left: 120px;
    border-top: 1px solid black;
    //z-index: 3;

    //AFFECTS NAIL
    border-radius: 100%;
    background-color: transparent;
    display: flex;
    justify-content: center;
    margin-bottom: -140px;
    .nail {
      background: gray;
      width: 10px;
      height: 10px;
      margin-top: -5px;
      border-radius: 100%;
      box-shadow: 3px 3px 5px 1px rgb(14, 13, 13);
    }
  }
}

@media (max-width: 960px) {
  .container__skills {
    padding-left: 1%;
    padding-right: 1%;
  }
}

@media (max-width: 622px) {
  .container__skills {
    min-height: calc(100vh - 245px);
  }
}
