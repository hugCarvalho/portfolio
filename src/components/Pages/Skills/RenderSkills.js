import React from "react";
import "./RenderSkills.scss";
import OneSkill, { RelatedSkills } from "./SkillCards";

function Skills() {
  return (
    <div className="container__skills">
      <section className="wrapper__all">
        {/* New Skill */}
        <div className="container__skill-card">
          <div className="wire">
            <span className="nail"></span>
          </div>
          <OneSkill classes="bg-javascript">JavaScript</OneSkill>
        </div>

        {/* New Skill */}
        <div className="container__skill-card">
          <div className="wire">
            <span className="nail"></span>
          </div>
          <OneSkill classes="bg-react">React</OneSkill>
        </div>

        {/* New Skill */}
        <div className="container__skill-card">
          <div className="wire">
            <span className="nail"></span>
          </div>
          <OneSkill classes="bg-redux">Redux</OneSkill>
        </div>

        {/* New Skill */}
        <div className="container__skill-card">
          <div className="box wire">
            <span className="nail"></span>
          </div>
          <OneSkill classes="bg-html-5">HTML</OneSkill>
        </div>

        {/* New Skill */}
        <div className="container__skill-card">
          <div className="wire">
            <span className="nail"></span>
          </div>
          <RelatedSkills
            skillOne="CSS"
            classesOne="bg-css3"
            skillTwo="Sass"
            classesTwo="bg-sass"
          />
        </div>

        {/* New Skill */}
        <div className="container__skill-card">
          <div className="box wire">
            <span className="nail"></span>
          </div>
          <RelatedSkills
            skillOne="Git"
            classesOne="bg-git"
            skillTwo="GitHub"
            classesTwo="bg-github special"
          />
        </div>
      </section>
    </div>
  );
}

export default Skills;
