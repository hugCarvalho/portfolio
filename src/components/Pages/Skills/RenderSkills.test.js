import React from "react";
import { render } from "@testing-library/react";
import RenderSkills from "./RenderSkills";

test("should match snapshot ", () => {
  const { container } = render(<RenderSkills />);
  expect(container).toMatchSnapshot();
});
