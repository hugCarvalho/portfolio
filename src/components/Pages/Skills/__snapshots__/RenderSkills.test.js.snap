// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should match snapshot  1`] = `
<div>
  <div
    class="Skills"
  >
    <main>
      <div
        class="container__skill-card"
      >
        <div
          class="wire"
        >
          <span
            class="nail"
          />
        </div>
        <div
          class="skill-box center"
        >
          <div
            class="logo bg-javascript"
          />
          <span>
            JavaScript
          </span>
        </div>
      </div>
      <div
        class="container__skill-card"
      >
        <div
          class="wire"
        >
          <span
            class="nail"
          />
        </div>
        <div
          class="skill-box center"
        >
          <div
            class="logo bg-react"
          />
          <span>
            React
          </span>
        </div>
      </div>
      <div
        class="container__skill-card"
      >
        <div
          class="wire"
        >
          <span
            class="nail"
          />
        </div>
        <div
          class="skill-box center"
        >
          <div
            class="logo bg-redux"
          />
          <span>
            Redux
          </span>
        </div>
      </div>
      <div
        class="container__skill-card"
      >
        <div
          class="box wire"
        >
          <span
            class="nail"
          />
        </div>
        <div
          class="skill-box center"
        >
          <div
            class="logo bg-html-5"
          />
          <span>
            Semantic HTML
          </span>
        </div>
      </div>
      <div
        class="container__skill-card"
      >
        <div
          class="wire"
        >
          <span
            class="nail"
          />
        </div>
        <div
          class="skill-box"
        >
          <div
            class="wrapper__logo-text"
          >
            <div
              class="logo bg-css3"
            />
            <span>
              CSS
            </span>
          </div>
          <div
            class="wrapper__logo-text"
          >
            <span>
              Sass
            </span>
            <div
              class="logo bg-sass"
            />
          </div>
        </div>
      </div>
      <div
        class="container__skill-card"
      >
        <div
          class="box wire"
        >
          <span
            class="nail"
          />
        </div>
        <div
          class="skill-box"
        >
          <div
            class="wrapper__logo-text"
          >
            <div
              class="logo bg-git"
            />
            <span>
              Git
            </span>
          </div>
          <div
            class="wrapper__logo-text"
          >
            <span>
              GitHub
            </span>
            <div
              class="logo bg-github special"
            />
          </div>
        </div>
      </div>
      <div
        class="container__skill-card"
      >
        <div
          class="box wire"
        >
          <span
            class="nail"
          />
        </div>
        <div
          class="skill-box center"
        >
          <div
            class="logo bg-jest"
          />
          <span>
            Jest
          </span>
        </div>
      </div>
      <div
        class="container__skill-card"
      >
        <div
          class="box wire"
        >
          <span
            class="nail"
          />
        </div>
        <div
          class="skill-box center"
        >
          <div
            class="logo bg-testing_library"
          />
          <span>
            <small>
              React Testing Library
            </small>
          </span>
        </div>
      </div>
    </main>
  </div>
</div>
`;
